[Site Isolation] Fix mouse event coordinates with scrolled frames

charliewolfe opened the pull request:
#### 20e46bea66daf37a1032bb1aa02dca3fdc656c84
<pre>
[Site Isolation] Fix mouse event coordinates with scrolled frames
<a href="https://bugs.webkit.org/show_bug.cgi?id=262905">https://bugs.webkit.org/show_bug.cgi?id=262905</a>
rdar://116690742

Reviewed by Alex Christensen.

Currently, mouse events passed to site-isolated frames will be offset if the target frame or target
frame’s ancestors are scrolled. We should convert to/from root view coordinates so that the scroll
position for each frame is considered.

Also change the parameter back on `documentPointForWindowPoint()` since that function should only be
used for local frames.

* LayoutTests/http/tests/site-isolation/mouse-events/resources/message-mouse-down-coordinates.html: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/resources/scroll-and-message-mouse-down-coordinates.html: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-mainframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-mainframe.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::documentPointForWindowPoint):
(WebCore::EventHandler::mouseEventDataForRemoteFrame):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
* Source/WebCore/page/EventHandler.h:

Canonical link: <a href="https://commits.webkit.org/269143@main">https://commits.webkit.org/269143@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1d68de82b9c037ffbf2882d2bfc443dddbca3fc9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21663 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/21946 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22706 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23526 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20055 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21904 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/26155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22193 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21221 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21890 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/26155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18758 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24380 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/26155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19611 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25908 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/26155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19820 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23767 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20307 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17298 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19635 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/19427 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/5169 "The change is no longer eligible for processing.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23861 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2690 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20226 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/page/EventHandler.h:
```diff
@@ -557,6 +559,8 @@ class EventHandler {
     bool canMouseDownStartSelect(const MouseEventWithHitTestResults&);
     bool mouseDownMayStartSelect() const;
+
+    std::optional<RemoteMouseEventData> mouseEventDataForRemoteFrame(const RemoteFrame&, const IntPoint&);
```
static
This doesn't use any member variables of the EventHandler

charliewolfe replied:
`m_frame`?

achristensen07 commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -1736,14 +1737,30 @@ void EventHandler::autoHideCursorTimerFired()
     return view ? view->windowToContents(windowPoint) : windowPoint;
 }
 
+std::optional<RemoteMouseEventData> EventHandler::mouseEventDataForRemoteFrame(const RemoteFrame& remoteFrame, const IntPoint& pointInFrame)
```
It would reduce more duplicate code if this took a const RemoteFrame* and a const MouseEvent&.  Then you could have the call sites look like this:

if (auto remoteMouseEventData = mouseEventDataForRemoteFrame(dynamicDowncast<RemoteFrame>(subframe), mouseEvent))

charliewolfe committed:
[Site Isolation] Fix mouse event coordinates with scrolled frames
https://bugs.webkit.org/show_bug.cgi?id=262905
rdar://116690742

Reviewed by Alex Christensen.

Currently, mouse events passed to site-isolated frames will be offset if the target frame or target
frame’s ancestors are scrolled. We should convert to/from root view coordinates so that the scroll
position for each frame is considered.

Also change the parameter back on `documentPointForWindowPoint()` since that function should only be
used for local frames.

* LayoutTests/http/tests/site-isolation/mouse-events/resources/message-mouse-down-coordinates.html: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/resources/scroll-and-message-mouse-down-coordinates.html: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-mainframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mouse-events/scrolled-mainframe.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::documentPointForWindowPoint):
(WebCore::EventHandler::mouseEventDataForRemoteFrame):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
* Source/WebCore/page/EventHandler.h:

Canonical link: https://commits.webkit.org/269143@main
